src/main/java/org/truffleruby/RubyContext.java

```diff
@@ -102,8 +102,9 @@ public RubyContext(RubyLanguage language, TruffleLanguage.Env env) {
         this.env = env;
 
         final OptionsBuilder optionsBuilder = new OptionsBuilder();
-        optionsBuilder.set(System.getProperties());
+        optionsBuilder.set(env.getOptions());
         optionsBuilder.set(env.getConfig());
+        optionsBuilder.set(System.getProperties());
```

Comment:
There's a problem with the options API in that it always gives you all the options, even the default ones, and you can't tell if something is default because it hasn't been set or it's just the same value as default. So options from the SDK must come first or they'd reset everything else. Then options from the polyglot engine, then system properties. It's documented behaviour in `options.md`.